### PR TITLE
feat: ssr demos

### PR DESCRIPTION
**Pull request by loks0n**
Server side auth tutorials for:
- [x] Nextjs
- [x] Nuxt
- [x] Astro

### CODE REVIEW

File: src/routes/docs/tutorials/nextjs-ssr-auth/+page.ts
Comment (gewenyu99) [inline src/routes/docs/tutorials/nextjs-ssr-auth/+page.ts:1]: Looks like you converted some files accidetally?

File: src/routes/docs/tutorials/nextjs-ssr-auth/step-1/+page.markdoc [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+---
+
+Appwrite takes away the stress of building and maintaining a backend. Appwrite helps implement authentication, databases, file storage, and respond to real-time events with **secure** APIs out of the box. 
+If you're a Next.js developer, the examples in this guide show you how Appwrite can help you add authentication to Next.js apps faster.
+
+# Before you start {% #before-you-start %}
+
+Even if you've never tried Appwrite, you will get an idea of what it'll feel like to build with Next.js and Appwrite.
+
+If you're inspired and wish to follow along, make sure you've followed [Start with Next.js](https://appwrite.io/docs/quick-starts/sveltekit) first.
```
Comment (gewenyu99) [inline src/routes/docs/tutorials/nextjs-ssr-auth/step-1/+page.markdoc:19]: Link is wrong

File: src/routes/docs/tutorials/nextjs-ssr-auth/step-2/+page.markdoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
+---
+layout: tutorial
+title: Create project
+description: Add authentication to a Next.js project using Appwrite.
+step: 2
+---
+
+
+Create a react project using [Next.js](https://kit.svelte.dev/docs/creating-a-project).
```
Comment (gewenyu99) [inline src/routes/docs/tutorials/nextjs-ssr-auth/step-2/+page.markdoc:9]: Link is wrong

File: src/routes/docs/tutorials/nextjs-ssr-auth/step-3/+page.markdoc [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+Create a function to build services you need in a file like `src/lib/server/appwrite.js` and **exporting the instances**.
+
+As part of the function, set the current user's session if they are logged in. This is done by accessing the session cookie from the request and calling the `setSession(session)` with the cookie value.
+
+```js
+// src/lib/server/appwrite.js
+import { Client, Account } from "node-appwrite";
+import { parseCookie } from "next/dist/compiled/@edge-runtime/cookies";
+
+export const SESSION_COOKIE = "a_session";
```
Comment (gewenyu99) [inline src/routes/docs/tutorials/nextjs-ssr-auth/step-3/+page.markdoc:19]: What does this do?

File: src/routes/docs/tutorials/nextjs-ssr-auth/step-3/+page.markdoc [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    get account() {
+      return new Account(client);
+    },
+  };
+}
+```
+
+`APPWRITE_KEY`, `NEXT_PUBLIC_APPWRITE_ENDPOINT` and `NEXT_PUBLIC_APPWRITE_PROJECT` are environment variables that are exported in your project's [.env file](https://nextjs.org/docs/app/building-your-application/configuring/environment-variables).
+
+You can get the values for these variables from the Appwrite console. The `PUBLIC_APPWRITE_ENDPOINT` is the endpoint of your Appwrite server, and the `PUBLIC_APPWRITE_PROJECT` is the ID of the project you want to use. The `APPWRITE_KEY` is an Appwrite API key with the necessary permissions to read and write to the database.
```
Comment (gewenyu99) [inline src/routes/docs/tutorials/nextjs-ssr-auth/step-3/+page.markdoc:44]:
```suggestion
You can get the values for these variables from the Appwrite console. The `PUBLIC_APPWRITE_ENDPOINT` is the endpoint of your Appwrite server, and the `PUBLIC_APPWRITE_PROJECT` is the ID of the project you want to use. The `APPWRITE_KEY` is an Appwrite API key with the necessary permissions to read and write accounts and sessions.
```

**Review by gewenyu99 — CHANGES_REQUESTED** [review]

File: src/routes/docs/tutorials/nextjs-ssr-auth/step-1/+page.markdoc (hunk 1 above)
Comment (loks0n) [reply src/routes/docs/tutorials/nextjs-ssr-auth/step-1/+page.markdoc:19]: fixed

File: src/routes/docs/tutorials/nextjs-ssr-auth/+page.ts
Comment (loks0n) [reply src/routes/docs/tutorials/nextjs-ssr-auth/+page.ts:1]: These are intentional, GH is just showing a weird changelog

File: src/routes/docs/tutorials/nextjs-ssr-auth/step-2/+page.markdoc (hunk 2 above)
Comment (loks0n) [reply src/routes/docs/tutorials/nextjs-ssr-auth/step-2/+page.markdoc]: Fixed

**Pull request merged**